pkg/client/observatorium/observability_config.go

```diff
@@ -80,6 +70,74 @@ type ObservabilityCloudWatchLoggingConfig struct {
 	configFilePath                string                                                      `validate:"-"`
 }
 
+type DataPlaneObservabilityOIDCCredentials struct {
+	ClientID     string `yaml:"client_id" validate:"required"`
+	ClientSecret string `yaml:"client_secret" validate:"required"`
+}
+
+type DataPlaneObservabilityOIDCConfiguration struct {
+	AuthorizationServer string                                 `yaml:"authorization_server" validate:"required"`
+	Realm               string                                 `yaml:"realm" validate:"required"`
+	Credentials         *DataPlaneObservabilityOIDCCredentials `yaml:"credentials" validate:"required,dive"`
```

Comment:
Is there some reason this is a pointer? I think we might be able to directly use `DataPlaneObservabilityOIDCCredentials`.